modules/microshift-cni.adoc

```diff
@@ -0,0 +1,271 @@
+
+microshift-ovs-init::
+The `microshift-ovs-init.service` is installed by the `microshift-networking` RPM package as a dependent systemd service to microshift.service. It is responsible for setting up the OVS gateway bridge.
+
+OVN containers::
+Two OVN-Kubernetes daemon sets are rendered and applied by {product-title}.
+
+* *ovnkube-master*
+Includes the `northd`, `nbdb`, `sbdb` and `ovnkube-master` containers.
+
+* *ovnkube-node*
+The ovnkube-node includes the OVN-Controller container.
++
+After {product-title} boots, the OVN-Kubernetes daemon sets are deployed in the `openshift-ovn-kubernetes` namespace.
+
+Packaging::
+OVN-Kubernetes manifests and startup logic are built into {product-title}. The systemd services and configurations included in `microshift-networking` RPM are:
+
+* `/etc/NetworkManager/conf.d/microshift-nm.conf` for NetworkManager.service
+* `/etc/systemd/system/ovs-vswitchd.service.d/microshift-cpuaffinity.conf` for ovs-vswitchd.service
+* `/etc/systemd/system/ovsdb-server.service.d/microshift-cpuaffinity.conf`
+* `/usr/bin/configure-ovs-microshift.sh` for microshift-ovs-init.service
+* `/usr/bin/configure-ovs.sh` for microshift-ovs-init.service
+* `/etc/crio/crio.conf.d/microshift-ovn.conf` for CRI-O service
+
+[id="microshift-config-OVN-K_{context}"]
+== Configuring OVN-Kubernetes
+An OVN-Kubernetes config file must be written to `/etc/microshift/ovn.yaml`. {product-title} will use default OVN-Kubernetes configuration values if an OVN-Kubernetes config file is not customized.
+
+.Default `ovn.yaml`config values:
+[source,yaml]
+----
+ovsInit:
+  disableOVSInit: false
+  gatewayInterface: "" <1>
+  externalGatewayInterface: "" <2>
+mtu: 1400
+----
+<1> Default value is an empty string, which means "not-specified." The CNI network provider auto-detects to interface with the default route.
+<2> Default value is an empty string, which means disabled.
+
+To customize your configuration, use the following table for reference to valid values that can be used in your `ovn.yaml` config file.
+
+.Supported optional OVN-Kubernetes configurations for {product-title}.
+
+[cols="5",options="header"]
+|===
+|Field
+|Type
+|Default
+|Description
+|Example
+
+|`ovsInit.disableOVSInit`
+|bool
+|false
+|Skip configuring OVS bridge `br-ex` in `microshift-ovs-init.service`
+|true <1>
+
+|`ovsInit.gatewayInterface`
+|Alpha
+|eth0
+|Ingress that is the API gateway
+|eth0
+
+|`ovsInit.externalGatewayInterface`
+|Alpha
+|eth1
+|Ingress routing external traffic to your services and pods inside the node
+|eth1
+
+|mtu
+|uint32
+|1400
+|MTU value used for the pods
+|1300
+|===
+
+<1> The OVS bridge is required. When `disableOVSInit` is true, OVS bridge `br-ex` must be configured manually.
+
+.Example `ovn.yaml` config file:
+[source, yaml]
+----
+ovsInit:
+  disableOVSInit: true
+  gatewayInterface: eth0
+  externalGatewayInterface: eth1
+mtu: 1300
+----
+
+[IMPORTANT]
```

Comment:
we have this not in line 140, remove either of them?

Reply:
@zshi-redhat I didn't understand this one. Let's get this on the next iteration.